code/modules/clothing/belts/polymorph_belt.dm

```diff
@@ -0,0 +1,152 @@
+/// Belt which can turn you into a beast, once an anomaly core is inserted
+/obj/item/polymorph_belt
+	name = "polymorphic field inverter"
+	desc = "This device can scan and store DNA from other life forms."
+	slot_flags = ITEM_SLOT_BELT
+	icon = 'icons/obj/clothing/belts.dmi'
+	icon_state = "polybelt_inactive"
+	worn_icon_state = "polybelt_inactive"
+	item_flags = NOBLUDGEON
+	/// Typepath of a mob we have scanned, we only store one at a time
+	var/stored_mob_type
+	/// Have we activated the belt?
+	var/active = FALSE
+	/// Our current transformation action
+	var/datum/action/cooldown/spell/shapeshift/polymorph_belt/transform_action
+
+/obj/item/polymorph_belt/Destroy(force)
+	QDEL_NULL(transform_action)
+	return ..()
+
+/obj/item/polymorph_belt/examine(mob/user)
+	. = ..()
+	if (stored_mob_type)
+		var/mob/living/will_become = stored_mob_type
+		. += span_notice("It contains digitised [initial(will_become.name)] DNA.")
+	if (!active)
+		. += span_warning("It requires a Bioscrambler Anomaly Core in order to function.")
+
+/obj/item/polymorph_belt/item_action_slot_check(slot, mob/user, datum/action/action)
+	return slot & ITEM_SLOT_BELT
+
+/obj/item/polymorph_belt/update_icon_state()
+	icon_state = active ? "polybelt" : "polybelt_inactive"
+	worn_icon_state = active ? "polybelt" : "polybelt_inactive"
+	return ..()
+
+/obj/item/polymorph_belt/attackby(obj/item/weapon, mob/user, params)
+	if (!istype(weapon, /obj/item/assembly/signaler/anomaly/bioscrambler))
+		return ..()
+	balloon_alert(user, "inserting...")
+	if (!do_after(user, delay = 3 SECONDS, target = src))
```

Comment:
You probably want to `return TRUE` after this to prevent afterattack (regardless of if the do_after succeeds)